src/box/memtx_engine.c

```diff
@@ -161,6 +161,13 @@ memtx_engine_recover_snapshot(struct memtx_engine *memtx,
 	int64_t signature = vclock_sum(vclock);
 	const char *filename = xdir_format_filename(&memtx->snap_dir,
 						    signature, NONE);
+	/*
+	 * Filename saved in static buffer, which reused later, so we need to
+	 * save filename in storage, which is local for this function.
+	 */
+	char name[PATH_MAX];
+	strncpy(name, filename, PATH_MAX);
+	name[PATH_MAX - 1] = '\0';
```

Comment:
You can use `strlcpy()`. It is defined here: https://github.com/tarantool/tarantool/blob/master/src/trivia/util.h#L466.

Reply:
Thanks! Fixed